parsec/core/mountpoint/manager.py

```diff
@@ -268,15 +268,19 @@ async def safe_unmount_all(self):
 
 
 async def cleanup_macos_mountpoint_folder(base_mountpoint_path):
-    # In case of a crash on macOS, workspaces don't unmount correctly and leave empty directories
-    # in the default mount folder. This function is used to clean these anytime a login occurs.
-    for dirs in os.listdir(base_mountpoint_path):
-        dir_path = str(base_mountpoint_path) + "/" + str(dirs)
-        stats = os.statvfs(dir_path)
+    # In case of a crash on macOS, workspaces don't unmount correctly and leave
+    # empty mountpoints or directories in the default mount folder. This
+    # function is used to unmount or delete these anytime a login occurs.
+    for mountpoint_name in os.listdir(base_mountpoint_path):
+        mountpoint_path = str(base_mountpoint_path) + "/" + str(mountpoint_name)
+        stats = os.statvfs(mountpoint_path)
```

Comment:
Cannot use `pathlib.Path.stat` instead ? (and especially the trio version `trio.Path.stat`, which guarantees us the trio loop won't be stuck if the stat is done on a folder that is still a valid mountpoint)